### PR TITLE
5 processing of time series

### PR DESCRIPTION
**Pull request by fredmontet**
This branch refers to feature #5 about the processing of TimeSeries. First parts of the issue have already been added to onTime but not the correlation processor.

This PR adds the correlation processor, its documentation and fixes a few details.

### CODE REVIEW

**Review by VincentMagn — COMMENTED** [review]
Here are my comments for "correlation.py". The notebook has not been reviewed yet, this will be done in a next review.

File: src/ontime/processors/registry/correlation.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import numpy as np
+
+from ...time_series import TimeSeries
+
+
+class Correlation:
+    """Correlation class handles correlation computation in a TimeSeries"""
+
+    @staticmethod
+    def process(ts, window):
```
[the remark this answers is not in the document]
Comment (VincentMagn) [reply src/ontime/processors/registry/correlation.py:11]: Apply to other function definitions. I mark them as "same" but it does not appear

[the remark this answers is not in the document]
Comment (fredmontet) [reply src/ontime/processors/registry/correlation.py:11]: Done

File: src/ontime/processors/registry/correlation.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    {
+                        "time": time,
+                        "var_a": current_df.index[row],
+                        "var_b": current_df.columns[col],
+                        "correlation": value,
+                    }
+                )
+
+        # Step 5: Convert the list of non-redundant items to a new DataFrame
+        result_df = pd.DataFrame(non_redundant_items)
```
[the remark this answers is not in the document]
Comment (fredmontet) [reply src/ontime/processors/registry/correlation.py:70]: Done

File: src/ontime/processors/registry/correlation.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        # Step 1: Set 'time' as the index
+        df.set_index("time", inplace=True)
+
+        # Step 2: Create a unique identifier for each (row_index, col_index) pair
+        # This will become the column names in the reshaped DataFrame
+        df["pair"] = df["var_a"] + "_" + df["var_b"]
+
+        # Step 3: Pivot the table so that each 'pair' becomes a column
+        # The values in the table will be the 'correlation' values
+        pivoted_df = df.pivot(columns="pair", values="correlation")
```
[the remark this answers is not in the document]
Comment (fredmontet) [reply src/ontime/processors/registry/correlation.py:91]: Done

**Comment by fredmontet** [comment]
@VincentMagn Ok to merge now?

**Pull request merged**